streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateTaskTest.java

```diff
@@ -129,6 +135,55 @@ public void shouldProcessRecordsForOtherTopic() throws Exception {
         assertEquals(0, sourceOne.numReceived);
     }
 
+    @Test(expected = StreamsException.class)
```

Comment:
We should not use `expected` here but `try { globalStateTask.update(...); fail(...): } catch(ignore) {}` pattern. Also, the `ConsumerRecord()` should be instantiated outside/before the try-catch blog.